### ISSUE TITLE
Long deploy time due to "Looking up discovery application name for instance" for non-Eureka enabled applications

### ISSUE THREAD

**Issue by ciphero** [spinnaker/spinnaker#6408]
### Issue Summary:
Unable to disable Eureka Discovery in "Monitor Disable Cluster" phase for non-Eureka enabled applications.
"Looking up discovery application name for instance" x100.

### Cloud Provider(s):
AWS

### Environment:
Spinnaker version 1.18.12
Deploying EC2 instance application with static ALB (managed by Terraform).
Application is not using any service registry (like Eureka).

### Feature Area:
Clouddriver/Eureka, Service Registry, Pipelines, Deploy

### Description:
AWS account has Eureka configured as Health Provider for Spinnaker but we have multiple applications, of which some use Eureka for service registry and others do not, so disabling Eureka health provider on AWS account basis in Spinnaker is not an option here.
While deploying non-Eureka enabled applications to AWS, "Monitor Disable Cluster" phase always pass through retry and timeout until the phase short-circuit causing a long delay in the deploy.
"Looking up discovery application name for instance..."
e.g.
`{"phase":"DISABLE_ASG","status":"Looking up discovery application name for instance i-xxx (attempt: 99)"},{"phase":"DISABLE_ASG","status":"Could not find application name in Discovery or AWS, short-circuiting…`

Enabling/Disabling "**_Consider only cloud provider health when executing tasks_**" and manually modifying pipeline JSON by defining "interestingHealthProviders" to exclude consideration of Eureka does not change deploy behaviour.

```
"healthCheckType":"ELB",
"interestingHealthProviders":["Amazon","TargetGroup"]
```

Expecting to be able to deploy non-Eureka enabled application to AWS account configured with Eureka as a health provider  without waiting on Eureka/AWS discovery retry timeout and phase short-circuit in "Monitor Disable Cluster" phase.

### Steps to Reproduce:
Deploy non-Eureka enabled application with ALB to AWS account configured with Eureka as a health provider.

### Additional Details:
In all essence this is a duplicate of closed issue: [2685](https://github.com/spinnaker/spinnaker/issues/2685)

**Deployment Logs:**
```
"cloudProvider":"aws",
"shrinkToSize":1,
"allowDeleteActive":true,
"deploy.server.groups":{"us-west-2”:[“xxx”]},
”kato.tasks":[{"id":"01F3D3188VZZDRVPH6AST6RA73",
"status":{"completed":false,
"failed":false,
"retryable":false},
"history":[{"phase":"ORCHESTRATION",
"status":"Initializing Orchestration Task"},
{"phase":"ORCHESTRATION","status":"Processing op: DisableAsgAtomicOperation"},
{"phase":"DISABLE_ASG","status":"Initializing Disable ASG operation for [us-west-2:xxx]…”},
{“phase":"DISABLE_ASG","status":"Disabling ASG ‘xxx’ in us-west-2..."},
{"phase":"DISABLE_ASG","status":"Deregistering instances from Target Groups..."},
{"phase":"DISABLE_ASG","status":"Deregistering instances from Target Groups..."},
{"phase":"DISABLE_ASG","status":"Marking ASG xxx as Disable with Discovery"},
{"phase":"DISABLE_ASG","status":"Looking up discovery application name for instance i-xxx (attempt: 0)"},
{"phase":"DISABLE_ASG","status":"Looking up discovery application name for instance i-xxx (attempt: 1)"}
{"phase":"DISABLE_ASG","status":"Looking up discovery application name for instance i-xxx (attempt: 2)”}
…
{"phase":"DISABLE_ASG","status":"Looking up discovery application name for instance i-xxx (attempt: 97)"},
{"phase":"DISABLE_ASG","status":"Looking up discovery application name for instance i-xxx (attempt: 98)"},
{"phase":"DISABLE_ASG","status":"Looking up discovery application name for instance i-xxx (attempt: 99)"},
{"phase":"DISABLE_ASG","status":"Could not find application name in Discovery or AWS, short-circuiting (asg: xxx, region: us-west-2)"},
{"phase":"DISABLE_ASG","status":"Finished Disabling ASG xxx.”},
{“phase":"DISABLE_ASG","status":"Finished Disable ASG operation for [us-west-2:xxx].”},
{“phase":"ORCHESTRATION","status":"Orchestration completed."},
```

**Comment by Ciphero** [comment]
bump

**Comment by spinnakerbot** [comment]
This issue hasn't been updated in 45 days, so we are tagging it as 'stale'. If you want to remove this label, comment:

> @spinnakerbot remove-label stale

**Comment by amiteshchandra** [comment]
Can we bump this issue? This is causing pipeline execution to either fail or takes extra 15-20 min if ELB is used for Eureka configured account in Spinnaker.
I tried with interestingHealthProviders but that also doesn't work.

I see in "EurekaSupportConfigurationProperties" file default 
retryMax=10
then how come it always retry 100 times.
Other property has value of this,
attemptShortCircuitEveryNInstances = 100
Any clue or suggestion?
This one I got from the slack discussion.
https://github.com/spinnaker/clouddriver/blob/8c377ef6be07278cd8a54448980f2b2065069a34/clouddriver-eureka/src/main/groovy/com/netflix/spinnaker/clouddriver/eureka/deploy/ops/EurekaSupportConfigurationProperties.groovy#L24

**Comment by Ciphero** [comment]
@spinnakerbot remove-label stale

**Comment by spinnakerbot** [comment]
This issue is tagged as 'stale' and hasn't been updated  in 45 days, so we are tagging it as 'to-be-closed'. It will be closed in 45 days unless updates are made. If you want to remove this label, comment:

> @spinnakerbot remove-label to-be-closed

**Comment by spinnakerbot** [comment]
This issue is tagged as 'to-be-closed' and hasn't been updated  in 45 days, so we are closing it. You can always reopen this issue if needed.
